src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -176,15 +176,74 @@ class ArnoldOutput : public IECore::RefCounted
 				{
 					continue;
 				}
+
+				std::string formattedData;
+				if( boost::starts_with( it->first.string(), "header:" ) )
+				{
+					if( const IECore::BoolData *data = IECore::runTimeCast<const IECore::BoolData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "int '%s' %i" ) % it->first.string().substr( 7 ) % int(data->readable()) ) );
+					}
+					else if( const IECore::FloatData *data = IECore::runTimeCast<const IECore::FloatData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "float '%s' %f" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::IntData *data = IECore::runTimeCast<const IECore::IntData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "int '%s' %i" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::StringData *data = IECore::runTimeCast<const IECore::StringData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::V2iData *data = IECore::runTimeCast<const IECore::V2iData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::V3iData *data = IECore::runTimeCast<const IECore::V3iData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::V2fData *data = IECore::runTimeCast<const IECore::V2fData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::V3fData *data = IECore::runTimeCast<const IECore::V3fData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::Color3fData *data = IECore::runTimeCast<const IECore::Color3fData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else if( const IECore::Color4fData *data = IECore::runTimeCast<const IECore::Color4fData>( it->second.get() ) )
+					{
+						customAttributes.push_back( boost::str( boost::format( "string '%s' %s" ) % it->first.string().substr( 7 ) % data->readable() ) );
+					}
+					else
+					{
+						IECore::msg( IECore::Msg::Warning, "IECoreArnold::Renderer", boost::format( "Can not convert '%s'. Must be of type string." ) % it->first.value() );
```

Comment:
"Must be of type string" is no longer true. Lets say `Cannot convert data "name" of type "type"`.